BiasAndFairnessModule/src/data_loader.py

```
@@ -0,0 +1,214 @@
+        Load the dataset based on configuration parameters.
+
+        Returns:
+            pd.DataFrame: Loaded dataset
+        """
+        if self.dataset_config.platform.lower() == "huggingface":
+            # Try to load the specified split, handle if unavailable
+            split = self.dataset_config.split
+            try:
+                dataset = load_dataset(self.dataset_config.source, split=split)
+            except Exception as e:
+                raise ValueError(f"Could not load split:'{split}' for dataset '{self.dataset_config.source}'. Error: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)